status-im/status-react
# feat: implement proper error handling for swaps

fixes #20342 
fixes #20271
fixes #21187

### Summary

This PR implements error handling for swap proposal errors.

<img src="https://github.com/user-attachments/assets/092576a2-f3ff-49ba-bd1c-99dd9e59e149" width="300"/>
<img src="https://github.com/user-attachments/assets/8de0237e-b0b6-440b-bd29-15bad19de499" width="300"/>
<img src="https://github.com/user-attachments/assets/1c9317c9-5ae0-4b18-a45e-11dc2134fb67" width="300"/>
<img src="https://github.com/user-attachments/assets/b6bb1174-4bbd-4a4b-9bad-ab9b92fa89c1" width="300"/>
<img src="https://github.com/user-attachments/assets/4bea11ed-d83b-40db-88a5-65fa09acf30a" width="300"/>
<img src="https://github.com/user-attachments/assets/99f3c4f8-9e35-4826-823d-88e6fa3e2778" width="300"/>

#### Platforms

- Android
- iOS

#### Areas that maybe impacted

##### Functional

- wallet / transactions

### Steps to test

- Open Status
- Login
- Go to wallet
- Select an account
- Select Swap option
- Select a token on Select asset to pay screen
- Enter a different amounts (low amount, max amount for ETH, an amount that exceeds the max amount)
- Wait for swap proposal or error to appear
- Verify the errors are shown correctly

status: ready
